upgrade/Upgrade-6.4.3.php

```diff
@@ -34,51 +36,35 @@
  *
  * @return bool
  */
-function upgrade_module_6_4_3($module)
+function upgrade_module_6_4_3(PayPal $module)
 {
-    $kit = new \PaypalAddons\services\Kit();
-    $baseDir = _PS_ROOT_DIR_ . '/modules/paypal/';
-    $dirs = [
-        '_dev',
-        'classes/APM',
-        'views/templates/apm',
-        'views/templates/admin/_partials/paypalBanner',
-        'views/templates/admin/_partials/form/fields',
-    ];
-    $files = [
-        'controllers/admin/AdminPayPalCustomizeCheckoutController.php',
-        'controllers/admin/AdminPayPalHelpController.php',
-        'controllers/admin/AdminPayPalInstallmentController.php',
-        'controllers/admin/AdminPayPalLogsController.php',
-        'controllers/admin/AdminPayPalPUIListenerController.php',
-        'controllers/admin/AdminPayPalSetupController.php',
-        'controllers/admin/AdminPaypalGetCredentialsController.php',
-        'views/templates/admin/customizeCheckout.tpl',
-        'views/templates/admin/help.tpl',
-        'views/templates/admin/installment.tpl',
-        'views/templates/admin/setup.tpl',
-        'views/templates/admin/_partials/accountSettingsBlock.tpl',
-        'views/templates/admin/_partials/blockPreviewButtonContext.tpl',
-        'views/templates/admin/_partials/block_info.tpl',
-        'views/templates/admin/_partials/carrierMap.tpl',
-        'views/templates/admin/_partials/ecCredentialFields.tpl',
-        'views/templates/admin/_partials/headerLogo.tpl',
-        'views/templates/admin/_partials/helperOptionInfo.tpl',
-        'views/templates/admin/_partials/mbCredentialsForm.tpl',
-        'views/templates/admin/_partials/pppCredentialsForm.tpl',
-        'views/templates/admin/_partials/roundingSettings.tpl',
-        'views/templates/admin/_partials/switchSandboxBlock.tpl',
-        'views/templates/admin/_partials/white-list.tpl',
-        'views/templates/admin/_partials/form/colorDescriptions.tpl',
-        'views/templates/admin/_partials/form/customizeStyleSection.tpl',
-        'views/templates/admin/_partials/form/sectionTitle.tpl',
-    ];
+    $baseDir = _PS_MODULE_DIR_ . 'paypal/';
+    $diagnosticConf = include _PS_MODULE_DIR_ . 'paypal/diagnostic.php';
+
+    if (empty($diagnosticConf[0]['stubs'][FileIntegrityStub::class])) {
+        return true;
+    }
 
-    foreach ($dirs as $dir) {
-        $kit->rrmdir($baseDir . $dir);
+    $stub = new FileIntegrityStub($diagnosticConf[0]['stubs'][FileIntegrityStub::class]);
+    $stub->setModule($module);
+    $response = $stub->getHandler()->handle();
+
+    if (empty($response['created'])) {
+        return true;
     }
+
+    $files = array_filter(
+        $response['created'],
+        function ($file) {
+            return !preg_match('/^config_[a-z]+\.xml$/', $file) && $file !== 'config.xml';
+        });
+
     foreach ($files as $file) {
-        $kit->unlink($baseDir . $file);
+        try {
+            unlink($baseDir . $file);
```

Comment:
Is there any risk that the module download an empty file and remove the all module ?

Reply:
unlink() can't remove directory

Reply:
Yes but all files not in the list. If the list is empty, all files will be removed.

Reply:
If an problem occurs on the stage of getting the list from Git, then none file is removed because script end on the line 53.
If problem occurs on the stage of preparing list to remove, one of possible scenarios is all files are removed.